flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeCasts.java

```diff
@@ -318,11 +318,17 @@ private static boolean supportsCasting(
             // cast between interval and exact numeric is only supported if interval has a single
             // field
             return isSingleFieldInterval(targetType);
+        } else if ((sourceType.is(CONSTRUCTED) || sourceType.is(STRUCTURED_TYPE))
+                && targetType.is(CHARACTER_STRING)) {
+            return supportsCollectionAndStructuredToStringCasting(sourceType, allowExplicit);
```

Comment:
can't this also just always be `true`? What casting does not support to string?

Reply:
We don't have any at the moment, at least looking at the matrix I wrote here https://github.com/apache/flink/pull/18813. Nevertheless, isn't it better to keep this code to avoid eventual future bugs with new types?

Reply:
Now we should be good to go, although I still had to do a couple of changes in this method to ensure the proper if-else branch :(